src/d3a/models/area/event_dispatcher.py

```diff
@@ -137,15 +136,13 @@ def create_agent_object(owner, higher_market, lower_market, is_spot_market):
                 else:
                     return OneSidedAgent(**agent_constructor_arguments)
             elif ConstSettings.IAASettings.MARKET_TYPE == 2:
-                return TwoSidedPayAsBidAgent(
-                    **agent_constructor_arguments,
-                    min_bid_age=ConstSettings.IAASettings.MIN_BID_AGE
-                )
-            elif ConstSettings.IAASettings.MARKET_TYPE == 3:
-                return TwoSidedPayAsClearAgent(
+                return TwoSidedAgent(
                     **agent_constructor_arguments,
                     min_bid_age=ConstSettings.IAASettings.MIN_BID_AGE
                 )
+            else:
+                assert False, f'Wrong market type setting flag ' \
+                              f'{ConstSettings.IAASettings.MARKET_TYPE}'
```

Comment:
Rather than relying on `assert False` to raise an Exception, I would recommend to explicitly raise a custom exception. For example:
```suggestion
                raise WrongMarketTypeException(
                    f"Wrong market type setting flag {ConstSettings.IAASettings.MARKET_TYPE}")
```